src/Controller/ProductController.php

```diff
@@ -226,7 +226,9 @@ public function productTeaserAction(Request $request, Factory $ecommerceFactory)
         if ($request->get('type') == 'object') {
             AbstractObject::setGetInheritedValues(true);
             $product = AbstractProduct::getById($request->get('id'));
-
+            if ($product instanceof Car && $product->getObjectType() === Car::OBJECT_TYPE_VIRTUAL_CAR) {
```

Comment:
Can we also add `className: 'Car',` property to renderlet editable on view template? so that other class objects cannot be assigned as well.